starfish/pipeline/algorithmbase.py

```diff
@@ -1,6 +1,39 @@
+class AlgorithmBase:
+    """
+    This is the base class of any algorithm that starfish exposes.
 
+    Subclasses of this base class are paired with subclasses of PipelineComponent. The subclasses of
+    PipelineComponent retrieve subclasses of the paired AlgorithmBase. Together, the two classes
+    enable starfish to expose a paired API and CLI.
 
-class AlgorithmBase:
+    Examples
+    --------
+
+    PipelineComponent: `starfish.image._segmentation.Segmentation(PipelineComponent)`
+
+    AlgorithmBase: `starfish.image._segmentation._base.SegmentationAlgorithmBase(AlgorithmBase)`
+
+    Implementing Algorithms:
+    - `starfish.image._segmentation.watershed.Watershed(SegmentationAlgorithmBase)`
+
+    This pattern exposes the API as follows:
+
+    `starfish.image.Segmentation.<implementing algorithm (Watershed)>`
+
+    and the CLI as:
+
+    `$> starfish segmentation watershed`
+
+    To create an entirely new algorithm class, like `Segmentation`, a new subclass of both
```

Comment:
This is confusing.  Segmentation is not an algorithm class in the programming sense of the word class.  It might make sense if you're talking about a class of algorithms in the non-programming sense, but then it just seems like we should use a non-ambiguous term.